### PR TITLE
Select all checkbox tiles view

### PR DESCRIPTION
**Pull request by AlexAndBear**
## Description
image

## How Has This Been Tested?

- test environment:
- test case 1:
- test case 2:
- ...

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests
- [ ] Documentation
- [ ] Maintenance (e.g. dependency updates or tooling)

## Open tasks:

- [ ] ...

### CODE REVIEW

File: packages/web-pkg/src/components/FilesList/ResourceTiles.vue [hunk 1]
```diff
@@ -1,46 +1,59 @@
...
+          </oc-list>
+        </oc-drop>
+      </div>
     </div>
     <oc-list class="oc-tiles oc-flex">
       <li
         v-for="resource in resources"
         :key="resource.id"
         class="oc-tiles-item has-item-context-menu"
+        v-memo="[resources]"
```
Comment (AlexAndBear) [inline packages/web-pkg/src/components/FilesList/ResourceTiles.vue:53]: @JammingBen @kulmann I added the v-memo directive, so the list gets only re-rendered when the resources do change (which is not the case when a resource gets selected) which makes the the selection quite fast with 500 tiles and fixes https://github.com/owncloud/web/issues/10578.

I couldn't spot any bad side effects

Comment (JammingBen) [reply packages/web-pkg/src/components/FilesList/ResourceTiles.vue:52]: Nice! Can/Should we use it for the files table as well? 🤔

File: packages/design-system/src/components/OcCheckbox/OcCheckbox.vue [hunk 2]
```diff
@@ -84,14 +84,6 @@ export default defineComponent({
...
       default: false
     },
-    /**
-     * Is the label of the Checkbox visually hidden?
-     **/
-    hideLabel: {
-      type: Boolean,
-      required: false,
-      default: false
-    },
```
Comment (JammingBen) [inline packages/design-system/src/components/OcCheckbox/OcCheckbox.vue:94]: Why did you remove this, didn't you just add it recently?

File: packages/web-pkg/src/components/FilesList/ResourceTiles.vue [hunk 3]
```diff
@@ -363,10 +380,31 @@ export default defineComponent({
       return resource.processing === true
     }
 
+    const disabledResources: ComputedRef<Array<Resource['id']>> = computed(() => {
```
Comment (JammingBen) [inline packages/web-pkg/src/components/FilesList/ResourceTiles.vue:383]: Can we name this `disabledResourceIds`? Otherwise it suggests full resources here. Then you could also remove the type annotation `ComputedRef<Array<Resource['id']>>` and let TS infer the type.

**Review by JammingBen — COMMENTED** [review]
Something messes with the right-click on tiles, it only works the first time for me.

File: packages/design-system/src/components/OcCheckbox/OcCheckbox.vue (hunk 2 above)
Comment (AlexAndBear) [reply packages/design-system/src/components/OcCheckbox/OcCheckbox.vue:94]: I added labelHidden prop, which is more a11y compatible than the implementation behind the hideLabel prop. I adjusted the other occurrences

File: packages/web-pkg/src/components/FilesList/ResourceTiles.vue (hunk 3 above)
Comment (AlexAndBear) [reply packages/web-pkg/src/components/FilesList/ResourceTiles.vue:383]: Sure this was just copy paste

File: packages/web-pkg/src/components/FilesList/ResourceTiles.vue (hunk 1 above)
Comment (AlexAndBear) [reply packages/web-pkg/src/components/FilesList/ResourceTiles.vue:52]: From my observations, the table doesn't get rerendered again when props change but the list in the resource tiles component did, so I don't think it's necessary

Comment (AlexAndBear) [reply packages/web-pkg/src/components/FilesList/ResourceTiles.vue:52]: > Something messes with the right-click on tiles, it only works the first time for me.

Aaaaaaaand that is the v-memo side effect urghs

**Comment by JammingBen** [comment]
> Something messes with the right-click on tiles, it only works the first time for me.

This is still an issue it seems like.

File: packages/web-pkg/src/components/FilesList/ResourceTiles.vue (hunk 1 above)
Comment (AlexAndBear) [reply packages/web-pkg/src/components/FilesList/ResourceTiles.vue]: Removed the v-memo, this is not working correctly and scope creep, need to investigate further 🕵️

**Review by JammingBen — APPROVED** [review]
